### PR TITLE
Introduce Ballerina package service

### PR DESCRIPTION
**Pull request by prabushi**
## Purpose
Introduce Ballerina package service with **metadata** and **components** APIs to retrieve data for VSCode package overview.

Fixes https://github.com/ballerina-platform/plugin-vscode/issues/99

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by nadeeshaan — COMMENTED** [review]
Test coverage is also low. Shall we add a couple of tests as well?

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/packages/BallerinaPackageServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+                }
+                metadata.setPath(project.get().sourceRoot().toString());
+                ProjectKind projectKind = project.get().kind();
+                if (projectKind != ProjectKind.SINGLE_FILE_PROJECT) {
+                    metadata.setPackageName(project.get().currentPackage().packageName().value());
+                }
+                metadata.setKind(projectKind.name());
+            } catch (Throwable e) {
+                String msg = "Operation 'ballerinaPackage/metadata' failed!";
+                this.clientLogger.logError(PackageContext.PROJ_PROJECT, msg, e, request.getDocumentIdentifier(),
```
[the remark this answers is not in the document]
Comment (prabushi) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/packages/BallerinaPackageServiceImpl.java]: Updated.

**Comment by prabushi** [comment]
> Test coverage is also low. Shall we add a couple of tests as well?

Added testcases.

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**